src/sonic-config-engine/config_samples.py

```diff
@@ -238,6 +244,7 @@ def generate_l2_config(data):
             # Ports in use should be admin up, unused ports default to admin down
             if port in downlinks or port in uplinks:
                 data['PORT'][port].setdefault('admin_status', 'up')
+                data['PORT'][port].setdefault('speed', '50000')
```

Comment:
>setdefault [](http://example.com/codeflow?start=35&length=10)

the l2_config is used for long time, why it is working but not yang compliant? Is it possible a yang bug?

Reply:
ref: [https://github.com/sonic-net/SONiC/wiki/L2-Switch-mode](https://github.com/sonic-net/SONiC/wiki/L2-Switch-mode)

Reply:
In general, `port_config.ini` always contains the speed column.
But the Unit test was using a port_config.ini doesn't contain speed https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-config-engine/tests/t0-sample-port-config.ini 
That's probably why we didn't hit yang issue for such long time.